File: tests/core/resources/diff/npm/class/right.js
/**
 * A guessing game.
 */

const readline = require('readline');

class Game {
  constructor(min = 1, max = 100) {
    this.min = min;
    this.max = max;
    // Rename this attribute
    this.value = this.getRandomInt(this.min, this.max);
    this.rl = readline.createInterface({
      input: process.stdin,
      output: process.stdout
    });
  }

  getRandomInt(min, max) {
    return Math.floor(Math.random() * (max - min + 1)) + min;
  }

  async askQuestion(query) {
    return new Promise((resolve) => {
      this.rl.question(query, resolve);
    });
  }

  async play() {
    console.log(`I'm thinking of a number between ${this.min} and ${this.max}!`);

    while (true) {
      try {
        const input = await this.askQuestion('Enter your guess: ');
        const guess = parseInt(input, 10);

        if (isNaN(guess)) {
          console.log('Please enter a valid integer.');
        } else if (guess < this.value) {
          console.log('Too low!');
        } else if (guess > this.value) {
          console.log('Too high!');
        } else {
          console.log('You won!');
          break;
        }
      } catch (error) {
        console.error(`Error: ${error.message}`);
        break;
      }
    }

    this.rl.close();
  }
}

(async () => {
  try {
    const game = new Game();
    await game.play();
  } catch (error) {
    console.error(`Failed to start the game: ${error.message}`);
    process.exit(1);
  }
})();

File: tests/core/resources/diff/pypi/class/right.py
"""
A guessing game.
"""

import random
import sys


class Game:
    def __init__(self, min=1, max=100):
        self.min = min
        self.max = max
        # Changed the name of this attribute
        self.value = random.randint(self.min, self.max)

    def play(self):
        print(f"I'm thinking of a number between {self.min} and {self.max}!")

        while True:
            try:
                guess = int(input("Enter your guess: "))
                
                if guess < self.value:
                    print("Too low!")
                elif guess > self.value:
                    print("Too high!")
                else:
                    print("You won!")
                    break
            except ValueError:
                print("Please enter a valid integer.")


def main() -> int:
    try:
        Game().play()
        return 0
    except Exception as e:
        sys.stderr.write(f"Error: {e}\n")
        return 1


if __name__ == "__main__":
    sys.exit(main())
